test/integration/install_test.go

```diff
@@ -637,12 +646,19 @@ func TestUpgradeHelm(t *testing.T) {
 		testutil.AnnotatedFatalf(t, "'helm upgrade' command failed",
 			"'helm upgrade' command failed\n%s\n%s", stdout, stderr)
 	}
+	TestHelper.WaitRollout(t, testutil.LinkerdDeployReplicasEdge)
 
 	vizChart := TestHelper.GetLinkerdVizHelmChart()
 	if stdout, stderr, err := TestHelper.HelmCmdPlain("upgrade", vizChart, "l5d-viz", vizArgs...); err != nil {
 		testutil.AnnotatedFatalf(t, "'helm upgrade' command failed",
 			"'helm upgrade' command failed\n%s\n%s", stdout, stderr)
 	}
+	expectedVizDeployments := testutil.LinkerdVizDeployReplicas
+	if TestHelper.ExternalPrometheus() {
+		delete(expectedVizDeployments, "prometheus")
+	}
```

Comment:
From [my testing](https://play.golang.org/p/_j-cgAYGOvW), `expectedVizDeployments` receives a reference to that map, so this delete will end up mutating `testutil.LinkerdVizDeployReplicas`. Apparently the go-way for this to clone the map using a loop...